src/formatDate.js

```diff
@@ -8,7 +8,44 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const slicedDate = date.split(fromFormat[3]);
+  const newSeparator = toFormat[3];
+
+  const dayIndex = fromFormat.indexOf('DD');
+  const monthIndex = fromFormat.indexOf('MM');
+  const yearIndex = Math.abs(dayIndex + monthIndex - 3);
+
+  const newDayIndex = toFormat.indexOf('DD');
+  const newMonthIndex = toFormat.indexOf('MM');
+  const newYearIndex = Math.abs(newDayIndex + newMonthIndex - 3);
```

Comment:
Similarly to the previous comment, finding newYearIndex using Math.abs(newDayIndex + newMonthIndex - 3) is not clear. It would be more readable to search for 'YYYY' or 'YY' in the toFormat array.